webgrid/__init__.py

```diff
@@ -189,10 +213,16 @@ def new_instance(self, grid):
                 if six.PY2 else inspect.getfullargspec(self.__init__).args)
 
         for argname in args:
-            if argname != 'self' and hasattr(self, argname) and \
-                    argname not in ('label', 'key', 'filter', 'can_sort'):
+            if argname != 'self' and argname not in (
+                'label', 'key', 'filter', 'can_sort', 'render_in', 'visible'
+            ) and hasattr(self, argname):
                 setattr(column, argname, getattr(self, argname))
 
+        # Copy underlying value of render_in and visible, in case they are
+        # lambdas that should be called per grid instance.
```

Comment:
The only issue I see with this plan is a potential performance problem. Renderers call the grid's `iter_columns`, on each row, which will force a new call on any columns that have these as lambdas. While that's not a huge deal for a web view with, say, 50-100 rows, that could really bog things down (depending on the efficiency of the lambda) for an XLSX with 100k records.

I think things are fine here, FWIW - the change needs to happen in the renderers. We're not passing row data to these lambdas, so they do not need to be evaluated per row. The renderers need to have a `columns` property that caches its list of columns, and pulls from `iter_columns` when that cache hasn't been set yet.
- I'd like to see a `Renderer` base class with that property
- Some of the generic stuff in the various renderers (`__init__`, `__call__`) can get moved there
- `Renderer.__init__` should check for an `init` method which takes no arguments, and if available, call it. The various renderers can use that to set up anything else they'd need.